### ISSUE TITLE
Add uint16 images' support for co-occurrence matrix

### ISSUE THREAD

**Issue by ThomasWalter** [scikit-image/scikit-image#2080]
## Description
The co-occurrence matrix implemented in feature/texture.py (skimage.feature.greycomatrix) does not accept images with more than 256 levels. This is reasonable for the calculation of Haralick features, but I think that it might be useful to remove this limitation for other use cases. 

As an example, I am currently working on an application, where I want to build a neighboring graph of several objects in an image. For this, I calculate a Voronoï tessellation (objects are the seeds), which partitions the image plane into labeled regions. Finding touching regions corresponds to identifying pairs of values with co-occurrence > 0. The following code does that: 

```
cooc = skimage.feature.greycomatrix(
    labels, [1],
    [0, np.pi/4, np.pi/2, 3*np.pi/4, np.pi, 5*np.pi/4, 3*np.pi/2, 7*np.pi/4],
    levels=labels.max() + 1
)

cooc = np.sum(cooc[:,:,0,:], axis=2)
cooc = cooc - np.diag(np.diag(cooc))
adjmat = cooc>0
```

Here `labels` is a label image (here an output of a Voronoï tessellation). Obviously, this only works if there are less than 256 objects in the image, due to the limitation of skimage.feature.greycomatrix.

**Comment by raamana** [comment]
Perhaps you fork it and disable the limitation?

**Comment by ThomasWalter** [comment]
Sure.

**Comment by stefanv** [comment]
@ThomasWalter Your use case sounds entirely reasonable to me.  I apologize: somehow this issue slipped by the members of the scikit-image team.  The first comment above is *not* by a member of the team and is also very atypical of the kind of response you can expect from us.  We always try to accommodate additions that make scikit-image more practical and easy to use.

/cc @jni

**Comment by ThomasWalter** [comment]
No problem and thank you and your team members for being so helpful!

**Comment by soupault** [comment]
Closed via #2095.
